### PR TITLE
EVG-14631 log error output before returning from git push

### PR DESCRIPTION
**Pull request by ablack12**
[EVG-14631](https://jira.mongodb.org/browse/EVG-14631)

### Description 
Sometimes we log exit code 128 with no context. This is likely some internal error, but we were returning before logging the error message that describes the issue, so we haven't been giving any information on if this is something users should be concerned with or not.

### CODE REVIEW

File: agent/command/git_push.go [hunk 1]
```diff
@@ -163,15 +163,14 @@ func (c *gitPush) pushPatch(ctx context.Context, logger client.LoggerProducer, p
 	cmd := jpm.CreateCommand(ctx).Directory(p.directory).Append(pushCommand).
 		SetOutputSender(level.Info, logger.Task().GetSender()).SetErrorWriter(stdErr)
 	if err := cmd.Run(ctx); err != nil {
+		errorOutput := stdErr.String()
```
Comment (ybrill) [inline agent/command/git_push.go:166]: looking at this again, maybe we could log stderr before we check the error? Something like 
```
err := cmd.Run(ctx)
[log stderr...]
if err != nil {
    return errors.Wrap(err, "can't push to remote")
}
```
this way stderr would get logged regardless of the exit code. 
I'm not certain if git would print things on stderr if we're exiting 0, but maybe we should handle it?

Comment (ablack12) [reply agent/command/git_push.go:166]: sure, I'm neutral

File: agent/command/git_push.go [hunk 2]
```diff
@@ -162,12 +162,13 @@ func (c *gitPush) pushPatch(ctx context.Context, logger client.LoggerProducer, p
...
 		SetOutputSender(level.Info, logger.Task().GetSender()).SetErrorWriter(stdErr)
-	if err := cmd.Run(ctx); err != nil {
-		errorOutput := stdErr.String()
-		if errorOutput != "" && p.token != "" {
-			errorOutput = strings.Replace(errorOutput, p.token, "[redacted oauth token]", -1)
-			logger.Execution().Error(errorOutput)
-		}
+	err := cmd.Run(ctx)
+	errorOutput := stdErr.String()
+	if errorOutput != "" && p.token != "" {
```
Comment (ybrill) [inline agent/command/git_push.go:167]: I noticed now that the way this was written if p.token isn't set we also won't log it. I don't _think_ it matters (I think it's always set) but I wanted to bring it to your attention.

**Review by ybrill — APPROVED** [review]

File: agent/command/git_push.go (hunk 2 above)
Comment (ablack12) [reply agent/command/git_push.go:167]: Thank you for looking at this more closely than I did haha

**Review by ybrill — APPROVED** [review]
it looks absolutely perfect 😄

**Pull request merged**